pkg/cel/cel.go

```diff
@@ -0,0 +1,141 @@
+package cel
+
+import (
+	"fmt"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/checker/decls"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	"github.com/google/cel-go/interpreter/functions"
+	exprpb "google.golang.org/genproto/googleapis/api/expr/v1alpha1"
+
+	"github.com/blang/semver/v4"
+)
+
+// PropertiesKey is the key for bundle properties map (input data for CEL evaluation)
+const PropertiesKey = "properties"
+
+type Constraint struct {
+	// Constraint message that surfaces in resolution
+	// This field is optional
+	Message string `json:"message" yaml:"message"`
+
+	// The cel struct that contraints CEL expression
+	// This field is required
+	Cel *Cel `json:"cel" yaml:"cel"`
+}
```

Comment:
This probably shouldn't be in the `cel` package right?

Reply:
If we change this package to `constraints` then I think it can be here. See above comment.